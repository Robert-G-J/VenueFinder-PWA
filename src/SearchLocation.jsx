import React, { Component } from "react";
import PropTypes from "prop-types";

class SearchLocation extends Component {
  render() {
    return (
      <div className="search-location__body">
        <input
          type="text"
          placeholder="Choose location, or..."
          className="search-location__input"
          onChange={e => this.props.onChange(e.target.value)}
        />
        <button className="search-location__button">Use current</button>
      </div>
    );
  }
}

SearchLocation.propTypes = {
  onChange: PropTypes.func,
  locationText: PropTypes.string
};

export default SearchLocation;
