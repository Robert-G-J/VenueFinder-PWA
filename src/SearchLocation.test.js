import React from "react";
import SearchLocation from "./SearchLocation";
import renderer from "react-test-renderer";

describe("The SearchLocation component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SearchLocation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
