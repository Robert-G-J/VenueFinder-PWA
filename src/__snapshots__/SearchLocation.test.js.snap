// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SearchLocation component renders correctly 1`] = `
<div
  className="search-location__body"
>
  <input
    className="search-location__input"
    onChange={[Function]}
    placeholder="Choose location, or..."
    type="text"
  />
  <button
    className="search-location__button"
    onClick={[Function]}
  >
    Use current
  </button>
</div>
`;
